.github/workflows/test.yml

```diff
@@ -0,0 +1,32 @@
+name: test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18, 20]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Node.js
```

Comment:
I think you can leverage containers as we also do in GitLab to avoid setting up Node.js yourself. 
Code snippet is here:

```
    strategy:
        matrix:
            node-version: [18, 20]

    container:
      image: node:${{ matrix.node-version }}
```

Reply:
Do you have a reference for this? While I was creating this CI file I was referring to official github documentation https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-nodejs which uses `actions/setup-node@v3`

Reply:
I found this here https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/running-jobs-in-a-container

Reply:
Resolved in https://github.com/initdsg/winston-cloudwatch/pull/1/commits/bf8d8c7b078d2088494b699c73404777e4b10894